api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -116,17 +116,29 @@ func (mw *azureManagedControlPlaneWebhook) ValidateCreate(ctx context.Context, o
 		)
 	}
 
-	if m.Spec.ControlPlaneEndpoint.Host != "" {
-		return field.Forbidden(
-			field.NewPath("Spec", "ControlPlaneEndpoint", "Host"),
-			controlPlaneEndpointErrorMessage,
-		)
+	ownerCluster := &clusterv1.Cluster{}
+	key := client.ObjectKey{
+		Namespace: m.Namespace,
+		Name:      m.Name,
```

Comment:
Is this making a similar assumption to the one that's being addressed in #3322? The owner references would probably be the easiest place to check for the Cluster's namespace/name, but that isn't set until after the AzureManagedControlPlane has been created. Otherwise it seems like we'd have to list all Clusters in all namespaces and see if any match this AzureManagedControlPlane and are paused.

Reply:
It's definitely true that the name of an AzureManagedControlPlane can be different from the name of the referencing cluster.

I think we can probably introspect each Cluster for Spec.ControlPlaneRef.Name == the name of the AzureManagedControlPlane; is that what you're thinking?

Reply:
Yeah, I was thinking of exactly that. Not sure there's any other way to correlate those right when the object is created.

Reply:
One other thought I had was to list with a field selector on the `spec.controlPlaneRef.*` fields but it looks like those aren't supported.

Reply:
And do we know for sure that `clusterctl move` will always create the `Cluster` before the `AzureManagedCluster`? The CAPI webhooks appear to allow either to be created first. If we have to account for when no matching Cluster exists yet here, I'm thinking we'd probably want to skip the `controlPlaneEndpoint` checks in that case to make sure we're not blocking a valid `clusterctl move`?

Reply:
Indeed maybe the most practical solution here is to remove these Create checks entirely. They were implemented defensively to protect the user from accidentally doing something wrong, but the outcome of "wrong" in this case is worst-case scenario a never-actually-manageable cluster, but best-case scenario the AKS controllers overwrite those user-provided values with the correct ones from the AKS API anyways, and so it literally has not effect. (We'd need to look into this to know for sure which outcome is the actual one. But either of those is probably not sufficiently catastrophic or likely to happen to justify all this additional webhook complexity.)

Thoughts?

cc @CecileRobertMichon @mtougeron

Reply:
> Indeed maybe the most practical solution here is to remove these Create checks entirely.

I tend to agree

> the AKS controllers overwrite those user-provided values with the correct ones from the AKS API anyways

are we sure this is still the case? I seem to remember we changed it to only set them when empty

Reply:
Indeed UT tells the tale:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/v1.8.1/api/v1beta1/azuremanagedcontrolplane_webhook_test.go#L1228-L1278

I'll submit a PR that removes create webhooks for this and AzureCluster and we can discuss there.